kotlin-native/NEW_MM.md

```diff
@@ -1,44 +1,36 @@
 # New memory model migration guide
 
-**NOTE**: _The new MM is still in an experimental stage. It's **not** production-ready._
+> The new memory model is Experimental. It's not production-ready and may be changed at any time.
+> Opt-in is required (see the details below), and you should use it only for evaluation purposes. We would appreciate your feedback on it in [YouTrack](https://youtrack.jetbrains.com/issue/KT-48525).
 
-In the new MM we are lifting restrictions placed on object sharing: there's no need to freeze objects to share them
-between threads.
+In the new memory model (MM), we're lifting restrictions on object sharing: there's no need to freeze objects to share them
+between threads anymore.
 
 In particular:
-* Top level properties can be accessed and modified by any thread without the need to use `@SharedImmutable`.
-* Objects passing through interop can be accessed and modified by any thread without the need to freeze them.
-* `Worker.executeAfter` will no longer require `operation` to be frozen, and `Worker.execute` will no longer require
-  `producer` to return an isolated object subgraph.
-
-Also, reference cycles containing `AtomicReference` and `FreezableAtomicReference` do not cause memory leaks anymore.
-
-A few caveats:
-* As with the previous MM, memory is not reclaimed eagerly: an object is reclaimed only when GC happens. This extends
-  to Swift/ObjC objects that crossed interop boundary into Kotlin/Native.
-* `AtomicReference` from `kotlin.native.concurrent` still requires freezing the `value`. `FreezableAtomicReference`
-  can be used instead, or, alternatively, `AtomicRef` from `atomicfu` can be used (**NOTE**: _`atomicfu` has not reached 1.x yet_).
-* `deinit` on Swift/ObjC objects (and the objects referred by them) will be called on a different thread if these objects
-  cross interop boundary into Kotlin/Native.
+* Top-level properties can be accessed and modified by any thread without using `@SharedImmutable`.
+* Objects passing through interop can be accessed and modified by any thread without freezing them.
+* `Worker.executeAfter` no longer requires `operation` to be frozen.
+* `Worker.execute` no longer requires `producer` to return an isolated object subgraph.
+* Reference cycles containing `AtomicReference` and `FreezableAtomicReference` do not cause memory leaks anymore.
+
+A few precautions:
+* As with the previous MM, memory is not reclaimed eagerly: an object is reclaimed only when GC happens. This extends to Swift/ObjC objects that crossed interop boundary into Kotlin/Native.
```

Comment:
Here we can omit that, but we always write the full name of the language (Objective-C in this case).